build-system/eslint-rules/preact.js

```diff
@@ -79,6 +84,67 @@ module.exports = {
       JSXFragment(node) {
         requirePreact(node);
       },
+
+      JSXSpreadAttribute(node) {
+        if (!staticTemplate) {
+          return;
+        }
+
+        context.report({
+          node,
+          message: [
+            'Static JSX Templates are required to use static attribute definitions',
+            'This prevents an issue with spread attributes accidentally overriding a "safe" attribute with user-provided data.',
+          ].join('\n\t'),
+        });
+      },
+
+      JSXOpeningElement(node) {
+        if (!staticTemplate) {
```

Comment:
Wouldn't this be an issue using Preact too?
```jsx
const Component = ({as: Comp = 'div', children}) =>
  <Comp>{children}</Comp>

<Component as={unsafeCouldBeScript}>
  {unsafeCouldBeScriptContent}
</Component>
```

Reply:
It could, but I think it's too late to change that for the Preact side since the `as: Comp` pattern is so common already. With this, we're trying to create very static looking trees, so the we don't require the same dynamism that Preact does.